### DISCUSSION TITLE
replace column. replace field option

### DISCUSSION THREAD

**Discussion by gstrat88** [Laravel-Backpack/community-forum#828] [Ideas]
Hello it would be nice if we could replace a column, and not modify it, as this does not work well with relathionships!
In that way it could be more easy to make modifications to a crud using setup from db!

app/Library/CrudPanel/Traits/Columns.php

```
public function replaceColumn($columnKey, $column = null)
    {
            $priority  = null;
            $columnsArray = $this->columns();
            if (isset($columnsArray[$columnKey])) {
                    $priority = $columnsArray[$columnKey]['priority'];
                    $columnsArray[$columnKey] = null;

            }

            if (is_array($column)) {
                    $column['name'] = $columnKey;
            } else {
                $column = ['name' => $columnKey];
            }

            $column['priority'] = $priority;


            $this->setOperationSetting('columns', $columnsArray);
            $this->addColumn($column);
    }
```

app/Library/CrudPanel/Traits/Fields.php

```
public function replaceField($fieldKey, $field = null) {
        $fields = $this->fields();

        if (isset($fields[$fieldKey])) {
                $fields[$fieldKey] = null;
        }

        if (is_array($field)) {
                $field['name'] = $fieldKey;
        } else {
                $field = ['name' => $fieldKey];
        }
        $this->setOperationSetting('fields', $fields);
        $this->addField($field);
    }

```

**Comment by jcastroa87** [comment]
Hello @gstrat88 I hope it goes well for you.

I don't understand if you have any problems with relationships. Can you please explain in detail so I can help. Thanks.

Cheers.

**Comment by gstrat88** [comment]
Scenario:
Model Article, category relatioship belongsTo using category_id field
In crud controller I am using the setupfromdb method to generate my form fields. Category_id field gets created as an integer field! So a replaceField method would be nice to have for such a case! Modify field works also, but you have to define a lot of things to make it work!

Also something irrelevent, where should i report bugs for the pro version?

**Reply by jcastroa87** [reply]
Hi @gstrat88 to report bugs about PRO, you can use: https://github.com/Laravel-Backpack/community-forum/issues

Cheers.

**Comment by o15a3d4l11s2** [comment]
@gstrat88, I think what you describe actually exists. After the fields are defined from the database, you can do one of two things:
- Use `modifyField` to update single (or few) of the field settings
- Use `addField` which will overwrite the specific field based on its name

As per your example you can do the following (in your Article CRUD)
```php
protected function setupCreateOperation(): void
{
    CRUD::setFromDb();

    CRUD::field('category_id')->type('relationship');
}
```
Note that you must have defined the relationship (in your Article model)
```php
public function category(): BelongsTo
{
    return $this->belongsTo(Category::class);
}
```
This will replace your `int` `category_id` field with a `relationship` one, which will work as expected.
